ElrondNetwork/api.elrond.com
# add parse enum pipe for collection and nft endpoints

## Type


## Problem setting
-  No validation for type filter 
  
## Proposed Changes
-  Add ParseOptionalEnumPipe 

## How to test
- nfts?type=SemiFungibleESDT -> should return a list of nfts
- nfts?type=SemiFungibleESDTTEST -> should return a proper validation message
